src/main/java/su/litvak/chromecast/api/v2/Channel.java

```diff
@@ -111,18 +111,27 @@ public void run() {
         @Override
         public void run() {
             while (!stop) {
+                CastChannel.CastMessage message = null;
                 try {
-                    CastChannel.CastMessage message = read();
+                    message = read();
                     if (message.getPayloadType() == CastChannel.CastMessage.PayloadType.STRING) {
-                        LOG.debug(" <-- {}", message.getPayloadUtf8());
+
                         final String jsonMSG = message.getPayloadUtf8().replaceFirst("\"type\"", "\"responseType\"");
+                        if (!jsonMSG.contains("responseType")) {
+                            LOG.warn(" <-- {Skipping}", jsonMSG);
+                            continue;
+                        }
                         Response parsed = jsonMapper.readValue(jsonMSG, Response.class);
                         if (parsed.requestId != null) {
                             ResultProcessor rp = requests.remove(parsed.requestId);
                             if (rp != null) {
                                 rp.put(parsed);
                             } else {
-                                LOG.warn("Unable to process request ID = {}, data: {}", parsed.requestId, jsonMSG);
+                                if (parsed.requestId != 0) {
+                                    // Status events are sent with a requestid of zero
+                                    // https://developers.google.com/cast/docs/reference/messages
+                                    LOG.warn("Unable to process request ID = {}, data: {}", parsed.requestId, jsonMSG);
```

Comment:
Reviewing this code, the response should be processed, since it is notifying us of a change in state. Maybe make `requestId 0' have a permanent result processor?